### PR TITLE
Removing uses of dir::topdown|dir::once

### PR DESCRIPTION
**Pull request by mjp41**
The semantics of `dir::topdown|dir::once` is quite complex.  This PR is an experiment to remove its use from rego-cpp. I believe it should be equivalent, but has split a few passes.

### CODE REVIEW

**Comment by mjp41** [comment]
* [ ] Update comments
* [ ] Separate passes in line with current style
* [ ] Tidy up well-formed definitions

File: include/rego/passes.h [hunk 1]
```diff
@@ -31,10 +31,12 @@ namespace rego
   PassDef explicit_enums();
   PassDef body_locals(const BuiltIns& builtins);
   PassDef value_locals(const BuiltIns& builtins);
-  PassDef compr();
+  PassDef compr1();
```
Comment (matajoh) [inline include/rego/passes.h:34]: I've been trying to keep the names vaguely related to what happens in the past in most cases. Perhaps you can change this to `rules_to_compr`?

File: include/rego/passes.h [hunk 2]
```diff
@@ -31,10 +31,12 @@ namespace rego
...
+  PassDef compr2();
   PassDef absolute_refs();
   PassDef merge_modules();
-  PassDef skips();
+  PassDef skips1();
```
Comment (matajoh) [inline include/rego/passes.h:38]: Similarly, could this be `datarule` or similar?

**Review by matajoh — CHANGES_REQUESTED** [review]
LGTM

**Review by matajoh — APPROVED** [review]

**Pull request merged**